cmd/dogstatsd/subcommands/start/command.go

```diff
@@ -103,12 +97,14 @@ func RunDogstatsdFct(cliParams *CLIParams, defaultConfPath string, defaultLogFil
 func start(cliParams *CLIParams, config config.Component, params *Params, server dogstatsdServer.Component) error {
 	// Main context passed to components
 	ctx, cancel := context.WithCancel(context.Background())
-	defer StopAgent(cancel)
+	var metaScheduler metadata.Scheduler
+	var dogstatsdStats http.Server
+	defer StopAgent(cancel, server, &metaScheduler, &dogstatsdStats)
 
 	stopCh := make(chan struct{})
 	go handleSignals(stopCh)
 
-	err := RunAgent(ctx, cliParams, config, params, server)
+	err := RunAgent(ctx, cliParams, config, params, server, &metaScheduler, &dogstatsdStats)
```

Comment:
Whereas it works well, it is more common to return the values instead of assignment them with pointers.

Reply:
FYI this is an outdated commit - ive since changed this code to use a struct that bundles the previously global dependencies together. I think this can go away once `MetaScheduler` and `DogstatdStats` are also components and provided at the constructor.